src/cocaine-app/yt_worker.py

```diff
@@ -0,0 +1,170 @@
+from logging import getLogger
+from yql.api.v1.client import YqlClient
+import time
+import datetime
+
+
+logger = getLogger('mm.planner')
+
+
+class YqlWrapper(object):
+    def __init__(self, cluster, token, attempts, delay):
+        self._cluster = cluster
+        self._token = token
+        self._attempts = attempts
+        self._delay = delay
+
+    def send_request(self, query, timeout=None):
+        """
+        Send request to YQL: ping verification + attempts retry
+        @return result dict on success, None otherwise
+        """
+        logger.debug('Send YQL request %s', query)
+        result = None
+
+        with YqlClient(db=self._cluster, token=self._token) as yql:
+            try:
+                if not yql.ping():
+                    logger.error("YQL ping failed %s %s", str(self._cluster), str(self._token))
+                    return None
+            except:
+                logger.exception("YQL ping excepted %s %s", str(self._cluster), str(self._token))
+                return None
+
+            start_time = time.time()
+
+            for attempt in xrange(0, self._attempts):
+                try:
+                    request = yql.query(query)
+                    request.run()
+
+                    logger.info("YQL query is running with id {}".format(request.operation_id))
+                    result = request.results
+                    end_time = time.time()
+                    if result.is_success:
+                        logger.info("YQL query succeeded and took {}".format(end_time-start_time))
+                        if timeout and (end_time - start_time > timeout):
+                            logger.error("YQL query start {} end {}. Exceed timeout {}".format(start_time, end_time, timeout))
+                            raise AssertionError("Timeout has exceeded on YQL op")
+                        return result
+
+                    logger.error("YQL failed #%d: %s[%d] (%s,%s)", attempt, result.status,
+                                 request.status_code, request.explain(), request.exc_info)
+                    for error in result.errors:
+                        logger.error("YQL query result error: %s", str(error))
+
+                    time.sleep(self._delay)
+                except:
+                    logger.exception("YQL request excepted %s", query)
+                    return result
+
+        logger.exception("YQL request attempts has exhausted (%s)", str(self._attempts))
+        return result
```

Comment:
Fixed in https://github.com/yandex/mastermind/pull/311/commits/808a99a19c77cd1a34d2ad6416390a670b6394eb